src/Spice86/Views/DebugWindow.axaml

```diff
@@ -18,455 +20,468 @@
-                                    </TextBlock>
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.CrtControllerVerticalBlankingStart, StringFormat='VerticalBlankingStart: {0}'}">
-                                    </TextBlock>
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.CrtControllerVerticalSyncStart, StringFormat='VerticalSyncStart: {0}'}">
-                                    </TextBlock>
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.CrtControllerVerticalTotal, StringFormat='VerticalTotal: {0}'}">
-                                    </TextBlock>
-
-                                </WrapPanel>
-                                <Label HorizontalAlignment="Center" FontWeight="Bold">CRT Controller</Label>
-                            </Grid>
-                            <Grid RowDefinitions="Auto,*" Margin="5">
-                                <Label HorizontalAlignment="Center" FontWeight="Bold">Renderer</Label>
-                                <WrapPanel Grid.Row="1" Orientation="Vertical">
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.RendererWidth, StringFormat='Width: {0}'}">
-                                    </TextBlock>
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.RendererHeight, StringFormat='Height: {0}'}">
-                                    </TextBlock>
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.RendererBufferSize, StringFormat='BufferSize: {0}'}">
-                                    </TextBlock>
-                                    <TextBlock TextAlignment="Right"
-                                               Text="{Binding VideoCard.LastFrameRenderTime, StringFormat='FrameRenderTime: {0}'}">
-                                    </TextBlock>
-                                </WrapPanel>
-                            </Grid>
-                        </WrapPanel>
-                    </TabItem>
-                </TabControl>
-            </Grid>
-            <TextBlock Text="To show anything, pause the emulator first." IsVisible="{Binding !IsPaused}" />
-        </Grid>
-    </ScrollViewer>
+                <Button Margin="5,0,0,0" VerticalAlignment="Bottom" Content="Force Update"
+                        Command="{Binding UpdateDataCommand}">
+                </Button>
+            </WrapPanel>
+        </Border>
+        <TextBlock Grid.Row="0" Grid.RowSpan="2" Text="To show anything, pause the emulator first." IsVisible="{Binding !IsPaused}" />
```

Comment:
I'd really like to keep seeing the video registers change while the game is running. Having to pause it is a dealbreaker for me.

Reply:
I'll fix this. This was to avoid reads while the emulator changes state (once I had an exception while showing the values of the DAC. In other words, reading the ArgbPalette while its length changed mid-read lead to an unhandled IndexOutOfRangeException).

I agree that seeing values withotu having to pause is valuable. I think I'll just catch and ignore IndexOutOfRangeException in the DebugViewModel code that updates the data presented.

Reply:
Done.